sycl/include/CL/sycl/id.hpp

```diff
@@ -16,6 +16,13 @@
 
 __SYCL_INLINE_NAMESPACE(cl) {
 namespace sycl {
+// Forward declarations
+namespace detail {
+template <typename TransformedArgType, int Dims, typename KernelType>
+struct RoundedRangeKernel;
```

Comment:
after replacing 'struct' with 'class' above, please also change other places as well (otherwise there may be warnings due to class vs struct inconsistency).